Parse/src/main/java/com/parse/OfflineQueryLogic.java

```diff
@@ -1020,37 +1021,38 @@ public int compare(T lhs, T rhs) {
       public Task<Object> then(Task<Void> task) throws Exception {
         if (container instanceof ParseObject) {
           // Make sure this object is fetched before descending into it.
-          return fetchIncludeAsync(container, null, db).onSuccess(new Continuation<Void, Object>() {
-            @Override
-            public Object then(Task<Void> task) throws Exception {
-              return ((ParseObject) container).get(key);
+          return fetchIncludeAsync(store, container, null, db)
+              .onSuccess(new Continuation<Void, Object>() {
+                @Override
+                public Object then(Task<Void> task) throws Exception {
+                  return ((ParseObject) container).get(key);
+                }
+              });
+            } else if (container instanceof Map) {
+              return Task.forResult(((Map) container).get(key));
+            } else if (container instanceof JSONObject) {
+              return Task.forResult(((JSONObject) container).opt(key));
+            } else if (JSONObject.NULL.equals(container)) {
+              // Accept JSONObject.NULL value in included field. We swallow it silently instead of
+              // throwing an exception.
+              return null;
+            } else {
+              return Task.forError(new IllegalStateException("include is invalid"));
             }
-          });
-        } else if (container instanceof Map) {
-          return Task.forResult(((Map) container).get(key));
-        } else if (container instanceof JSONObject) {
-          return Task.forResult(((JSONObject) container).opt(key));
-        } else if (JSONObject.NULL.equals(container)) {
-          // Accept JSONObject.NULL value in included field. We swallow it silently instead of
-          // throwing an exception.
-          return null;
-        } else {
-          return Task.forError(new IllegalStateException("include is invalid"));
-        }
-      }
-    }).onSuccessTask(new Continuation<Object, Task<Void>>() {
-      @Override
-      public Task<Void> then(Task<Object> task) throws Exception {
-        return fetchIncludeAsync(task.getResult(), rest, db);
-      }
-    });
+          }
+        }).onSuccessTask(new Continuation<Object, Task<Void>>() {
+          @Override
+          public Task<Void> then(Task<Object> task) throws Exception {
+            return fetchIncludeAsync(store, task.getResult(), rest, db);
+          }
+        });
   }
 
   /**
    * Makes sure all of the objects included by the given query get fetched.
    */
-  /* package */ <T extends ParseObject> Task<Void> fetchIncludesAsync(
-      final T object,
+  /* package */ static <T extends ParseObject> Task<Void> fetchIncludesAsync(
+      final OfflineStore store, final T object,
```

Comment:
Nit: same with previous suggestion, if parameters list is longer than limitation, each parameter should use a new line.